docs/src/contributing/git-workflow.md

```diff
@@ -51,36 +99,173 @@ If the branch exists in a triton/neptune official repo, (as opposed to a persona
+1.2.0-rc0       (release candidates start at 1, not 0)
+```
+
+note: for git tags, a "v" is prefixed, as per git convention.
+
+Our release process treats these as:
+
+**major**: bumped any time there are:
+
+1. significant consensus or p2p layer changes. eg hard or soft-fork or breaking protocol change.
+2. breaking library API changes.
+3. breaking database changes.
+
+resets minor and patch to 0.
+
+**minor**: bumped any time changes from master are included in a release candidate.
+
+resets patch to 0.
+
+**patch**: bumped any time changes from a release branch are included in a release.
+
+resets candidate to "-rc1".
+
+**candidate**: added/bumped when a new public testnet is created for a candidate release.
+
+*Note that the optional suffix is supported by cargo and related tools.*
+
+## typical release
+
+The normal/typical release flow would be:
+
+1. normal changes (not consensus) are introduced in master.
+2. a release candidate is tagged from master, eg `v0.3.0-rc1` and built.
+3. a testnet is created for `v0.3.0-rc1`
+4. if significant problems are found then a new branch `v0.3.0-rc2` is created at `v0.3.0-rc1`, or possibly from master.  commit(s) are added. When ready, the final commit in the branch is tagged with `v0.3.0-rc2`, another testnet created, and so on.
+5. fixes for the release-candidate should also be applied to master, if applicable.
+6. After two weeks, or when team decides, the latest release-candidate tag is also tagged with `v0.3.0`, signifying an actual release.
+7. The release is performed.
+
+*note that by the time the release actually occurs, master may have diverged significantly.*
+
+The release flow for a major version change is essentially the same except that the major version number is bumped and the minor version returns to 0.
+
+
+## Release Candidate Protocol  (-rc1, -rc2, etc)
+
+ - determine source branch and new version.  (see visualization above.)
+ - checkout the source branch and ensure working directory is clean.
+ - Set the version in `Cargo.toml` (including the -rcX suffix).
+ - Ensure that source branch (eg master) builds against crates that live on
+   [crates.io](https://crates.io). In particular, no dependencies on github
+   repositories or revisions.
+ - review `README.md` to ensure the installation instructions are up-to-date.
  - Ensure that all tests pass.
- - Bump the version in `Cargo.toml`
- - Create a commit with the subject line `v0.0.6` (or watever the new version number is) and in the body list all the changes.
- - Push to `master` on github.
- - Add a tag marking the current commit with the version:
-   - `git tag v0.0.6` (or whatever the next version is)
+ - Create a commit with the subject line `v0.3.0-rc1` (or whatever the new version number is) and in the body list all the changes.
+ - Push present branch to github.
+ - Add a tag marking the current commit version:
+   - `git tag v0.3.0-rc1` (or whatever the next version is)
+   - `git push --tags`.
```

Comment:
I don't recommend using this command, since it pushes _all_ tags. See also [this](https://stackoverflow.com/questions/5195859/how-do-you-push-a-tag-to-a-remote-repository-using-git#comment40809114_5195913) stackoverflow comment.

Instead, let's go with `git push origin tag v0.3.0-rc1`.

Reply:
good point.  yeah, this was just keeping (copy/paste) what was already written so I didn't put thought into it.  will change.

Reply:
addressed in 66b171787d18c6ad4aaf45854923545d15666d28